x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/AbstractSqlQueryRequest.java

```diff
@@ -242,6 +248,30 @@ public ActionRequestValidationException validate() {
                     validationException);
             }
         }
+        if (runtimeMappings != null) {
+            // Perform a superficial check for runtime_mappings structure. The full check cannot happen until the actual search request
+            validationException = validateRuntimeMappings(runtimeMappings, validationException);
+        }
+        return validationException;
+    }
+
+    private static ActionRequestValidationException validateRuntimeMappings(Map<String, Object> runtimeMappings,
```

Comment:
Would it be possible to extract this in `ql` module?

Reply:
Unfortunately (and frustratingly), no. `sql-action` has almost no dependencies on other projects. Which is a good thing for the end user, but it brings some limitations in the code reusability and organization  area.